File: 프로그래머스/1/42889. 실패율/실패율.py
def solution(N, stages):
    sort = []
    for n in range(1,N+1):
        tmp = [i for i in stages if i>=n]
        if tmp.count(n)>0:
            sort.append([tmp.count(n)/len(tmp),n])
        else:
            sort.append([0,n])
    sort = sorted(sort,key=lambda x:(-x[0],x[1]))
    return [n for _,n in sort]

File: 프로그래머스/1/42889. 실패율/실패율.cpp
#include <string>
#include <algorithm>
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(int N, vector<int> stages) {
    vector<int> answer;
    vector<double> rate;
    int sum = 0;
    int num = 0;
    for(int i = 0; i < N; i++){
        num = count(stages.begin(), stages.end(), i+1);
        if(stages.size()==sum)
            rate.push_back(0);
        else
            rate.push_back((double)num/(stages.size()-sum));
        sum += num;
    }
    vector<int> idx;
    for(int i = 0; i < N; i++){
        idx.push_back(i+1);
    }
    sort(idx.begin(), idx.end(), [&rate](int a, int b){
        if(rate[a-1] == rate[b-1]) return a < b;
        return rate[a-1] > rate[b-1];
    });
    return idx;
}